portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +32,50 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+    
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Comment> comments = new ArrayList<>();
+
+    String commentLimitString = request.getParameter("limit");
+    int commentLimit = Integer.parseInt(commentLimitString);
+    if (commentLimit <= 0) commentLimit = 5;
+    for (Entity entity : results.asIterable()) {
+      String text = (String) entity.getProperty("text");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Comment comment = new Comment(text, timestamp);
+      comments.add(comment);
+      if (--commentLimit == 0) {
+        break;
+      }
+    }
+
+    Gson gson = new Gson();
+    String json = gson.toJson(comments);
+
     response.setContentType("text/html;");
```

Comment:
I think this should be "application/json;" now.

Reply:
Done.